---
title: "miR-532 targets"
author: "Patrick Lac"
date: "April 25, 2020"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, message = F, warning = F)
```

```{r libraries}
library(tidyverse)
library(dplyr)
library(RobustRankAggreg)
library(gprofiler2)
library(plotly)
library(enrichR)
```

```{r data}
AMLPMP_corrgenes <- read.csv("F:/Git/PLac Analyses/miR-532 project/miR-532/miR-532 targets/Data/Gene_corr_miR-532_AMLPMP.csv")
TCGA_corrgenes <- read.csv("F:/Git/PLac Analyses/miR-532 project/miR-532/miR-532 targets/Data/Gene_corr_miR-532_TCGA.csv")
```

Now let's prepare our list of genes for input into RankAggreg 

In rankAggreg, the algorithm looks at how each item is positioned in the ranked lists and compares this to the baseline case where the  lists are randomly shuffled. 

A P-value is calculated, showing how it is positioned in the aggregated list compared to chance. This P-value is used both for re-ranking the items and deciding their significance. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278763/)

```{r rename column}
colnames(AMLPMP_corrgenes)[1] <- "genes"
colnames(TCGA_corrgenes)[1] <- "genes"
```

```{r downreg_genes}
AMLPMP_downreg <- filter(AMLPMP_corrgenes, AMLPMP_corrgenes$estimate < 0)
TCGA_downreg <- filter(TCGA_corrgenes, TCGA_corrgenes$estimate < 0)
```


``` {r common_genes}
AMLPMP_shared <- filter(AMLPMP_downreg, AMLPMP_downreg$genes %in% TCGA_downreg$genes)
TCGA_shared <- filter(TCGA_downreg, TCGA_downreg$genes %in% AMLPMP_downreg$genes)
```

From the entire list of significantly correlated genes, 146 and 139 genes are negatively correlated with miR-532 expression in the AML-PMP and TCGA datasets respectively. 116 are common between the datasets.

```{r lists}
AMLPMP_ordered <- AMLPMP_shared[order(AMLPMP_shared$neg_corr_rank),] %>%
  pull(var=1) %>%
  as.matrix(nrow=1)

TCGA_ordered<- TCGA_shared[order(TCGA_shared$neg_corr_rank),] %>%
  pull(var=1) %>%
  as.matrix(nrow=1)

AMLPMP_anticorr <- AMLPMP_ordered[1:116,1]
TCGA_anticorr <- TCGA_ordered[1:116,1]
```

``` {r aggregateranks}
glist <- list(AMLPMP_anticorr, TCGA_anticorr)
finalrank <- aggregateRanks(glist = glist)
finalrank

rank.RRA <- aggregateRanks(glist = glist, method= "RRA")
rank.RRA

rank.stuart <- aggregateRanks(glist = glist, method= "stuart")
rank.stuart
```



Now let's convert the results into csv files:
``` {r csv}
write.csv(finalrank, file = "Aggregated_miR-532_targets(RRA).csv")
write.csv(rank.stuart, file = "Aggregated_miR-532_targets(Stuart).csv")
```

### Now let's look at the gene ontology 
Beginning by making a list from the aggregated gene rankings
```{r GO_genelist}
GO_list <- rank.RRA[, 1]
```

Now let's look perform the GO analysis

```{r GO_analysis}
GO_targets <- gost(GO_list, organism ="hsapiens")
GO_targets
```

Now we can try to visualize the results

```{r plot_GO}
GO_plot <- gostplot(GO_targets, interactive = T)
GO_plot
```

#### Now let's try using Enrichr

``` {r enrichr_532}
listEnrichrDbs()
enrichr_db <- listEnrichrDbs()
enrichr_db <- c("GO_Molecular_Function_2018", "GO_Cellular_Component_2018", "GO_Biological_Process_2018", "KEGG_2019_Human", "ChEA_2016", "ENCODE_TF_ChIP-seq_2015", "Transcription_Factor_PPIs", "ENCODE_Histone_Modifications_2015" ,"WikiPathways_2019_Human", "Reactome_2016", "PPI_Hub_Proteins", "Cancer_Cell_Line_Encyclopedia", "Human_Gene_Atlas", "ProteomicsDB_2020", "Mouse_Gene_Atlas", "CCLE_Proteomics_2020")

miR532_targets_enrichr <- enrichr(as.character(GO_list), enrichr_db)
printEnrich(miR532_targets_enrichr, "miR532_target_enrichr.txt" , sep = "\t", columns = c(1:9))
```

```{r enrichr_results}
MF_miR532 <- miR532_targets_enrichr[["GO_Molecular_Function_2018"]]
MF_miR532

CC_miR532 <- miR532_targets_enrichr[["GO_Cellular_Component_2018"]]
CC_miR532

BP_miR532 <- miR532_targets_enrichr[["GO_Biological_Process_2018"]]
BP_miR532

KEGG_miR532 <- miR532_targets_enrichr[["KEGG_2019_Human"]]
KEGG_miR532

ChEA_miR532 <- miR532_targets_enrichr[["ChEA_2016"]]
ChEA_miR532

EncodeTFCHIP_miR532 <- miR532_targets_enrichr[["ENCODE_TF_ChIP-seq_2015"]]
EncodeTFCHIP_miR532

TF_PPI_miR532 <- miR532_targets_enrichr[["Transcription_Factor_PPIs"]]
TF_PPI_miR532

histonemodification_miR532 <- miR532_targets_enrichr[["ENCODE_Histone_Modifications_2015"]]
histonemodification_miR532

wikipathways_miR532 <- miR532_targets_enrichr[["WikiPathways_2019_Human"]]
wikipathways_miR532

Reactome_miR532 <- miR532_targets_enrichr[["Reactome_2016"]]
Reactome_miR532

Protein_PPI_miR532 <- miR532_targets_enrichr[["PPI_Hub_Proteins"]]
Protein_PPI_miR532

HumanGeneAtlas_miR532 <- miR532_targets_enrichr[["Human_Gene_Atlas"]]
HumanGeneAtlas_miR532

MouseGeneAtlas_miR532 <- miR532_targets_enrichr[["Mouse_Gene_Atlas"]]
MouseGeneAtlas_miR532

Proteomics_miR532 <- miR532_targets_enrichr[["ProteomicsDB_2020"]]
Proteomics_miR532

CCLE_miR532 <- miR532_targets_enrichr[["Cancer_Cell_Line_Encyclopedia"]]
CCLE_miR532

CCLE_Proteomics_miR532 <- miR532_targets_enrichr[["CCLE_Proteomics_2020"]]
CCLE_Proteomics_miR532
```
